kod/object/active/holder/nomoveon/battler.kod

```diff
@@ -1383,6 +1451,862 @@ messages:
+            {
+               Send(self,@ResetPlayerFlagList,#who=self);
+            }
+         }
+      }
+
+      if IsClass(self,&Player)
+      {
+         % Also remove from passive improvement list.
+         for i in plPassiveImprovement
+         {
+            if IsClass(First(i),class)
+            {
+               plPassiveImprovement = DelListElem(plPassiveImprovement,i);
+            }
+         }
+      }
+
+      return removedSomething;
+   }
+
+   AddAttackModifier(what = $)
+   "Adds <what> to plAttack_modifiers, which can modify attack "
+   "strength and damage."
+   {
+      plAttack_modifiers = Cons(what,plAttack_modifiers);
+      
+      if IsClass(self,&Player)
+      {
+         Post(self,@DrawOffense);
+      }
+
+      return;
+   }
+
+   RemoveAttackModifier(what = $)
+   "Removes <what> from plAttack_modifiers."
+   {
+      local i;
+
+      for i in plAttack_modifiers
+      {
+         if i = what
+         {
+            plAttack_modifiers = DelListElem(plAttack_modifiers,i);
+            
+            if IsClass(self,&Player)
+            {
+               Post(self,@DrawOffense);
+            }
+
+            return;
+         }
+      }
+
+      return;
+   }
+
+   AddDefenseModifier(what = $)
+   "Adds <what> to plDefense_modifiers, which can modify defense strength "
+   "and damage."
+   {
+      plDefense_modifiers = Cons(what,plDefense_modifiers);
+      
+      if IsClass(self,&Player)
+      {
+         Post(self,@DrawDefense);
+         Post(self,@DrawArmor);
+      }
+
+      return;
+   }
+
+   RemoveDefenseModifier(what = $)
+   "Removes <what> from plDefense_modifiers."
+   {
+      local i;
+
+      for i in plDefense_modifiers
+      {
+         if i = what
+         {
+            plDefense_modifiers = DelListElem(plDefense_modifiers,i);
+            
+            if IsClass(self,&Player)
+            {
+               Post(self,@DrawDefense);
+               Post(self,@DrawArmor);
+            }
+            
+            return;
+         }
+      }
+      
+      Debug(self,"Tried to remove defense modifier",what,"but not in list",
+               plDefense_modifiers);
+
+      return;
+   }
+
+   AddPoison(strength = 0, bVigorLoss = TRUE)
+   {
+      if ptPoison = $
+      {
+         ptPoison = CreateTimer(self,@PoisonTimer,10000);
+      }
+
+      piPoisonStrength = piPoisonStrength + strength;
+
+      if bVigorLoss
+         AND IsClass(self,&Player)
+      {
+         piVigor = piVigor - 20;
+         Send(self,@NewVigor);
+      }
+
+      return;
+   }
+
+   RemovePoison(strength = 0)
+   {
+      piPoisonStrength = piPoisonStrength - strength;
+
+      return;
+   }
+
+   PoisonTimer()
+   {
+      ptPoison = $;
+
+      % For safety's sake
+      if piPoisonStrength <= 0
+      {
+         piPoisonStrength = 0;
+
+         return;
+      }
+
+      if poOwner <> $
+      {
+         ptPoison = CreateTimer(self,@PoisonTimer,10000);
+      }
+
+      if poOwner = $
```

Comment:
Move this check above ptPoison = CreateTimer(self,@PoisonTimer,10000);, then you can remove the poOwner <> $ check there as it won't be needed.